Doc Js Driver: Basic usage examples

michielmulders opened the pull request.

kremalicious commented:
hey @michielmulders, that's cool, thanks! But the file permission changes need to be somehow removed from this PR. You changed the permission of each and every file in this repo from `644` to `755`. 

Maybe revert via git locally, add your changes back on top again and force push to this branch. Or just change all permissions back and commit that change, which should remove it from this PR too. Let me know if you need help with that, git can get tricky with accidental published changes 🥃

TimDaub commented on docs/source/usage.rst:
```diff
@@ -0,0 +1,364 @@
+
+This example guides you through creating and transferring an asset.
+We walk through the code explaining its use, some pieces are left out
+because they have no real use (e.g. defenition of global variable)
```
`definition`

TimDaub commented on docs/source/usage.rst:
```diff
@@ -0,0 +1,364 @@
+
+Getting Started
+---------------
+We begin by creating an object of BigchainDB driver:
```
`We begin by importing the BigchainDB driver`

TimDaub commented on docs/source/usage.rst:
```diff
@@ -0,0 +1,364 @@
+Create Connection With BigchainDB
+---------------------------------
+
+A simpel connection with BigchainDB can be established like this.
```
`simple`

TimDaub commented on docs/source/usage.rst:
```diff
@@ -0,0 +1,364 @@
+.. code-block:: js 
+
+	let bdb = new driver.Connection('https://test.ipdb.io/api/v1/', { 
+		app_id: 'dgi829l9',
```
I hope these are not your actual keys to use the test net. If so please remove them from this PR.

michielmulders replied:
No, fake keys! :)

michielmulders replied:
But correct lenght

TimDaub replied:
👍

TimDaub commented on docs/source/usage.rst:
```diff
@@ -0,0 +1,364 @@
+   		metadata,
+
+		// A transaction needs an output
+		[ driver.Transaction.makeOutput(
```
I think you should explain between l104 - l107 what inputs and outputs in a BigchainDB transaction are.

TimDaub commented on docs/source/usage.rst:
```diff
@@ -0,0 +1,364 @@
+one day, she meets Bob, who is interested in acquiring her bicycle. The timing
+is good for Alice as she had been wanting to get a new bicycle.
+
+To transfer the bicycle (asset) to Bob, Alice must consume the transaction in
```
`Alice must consume the transaction's output`

TimDaub commented on docs/source/usage.rst:
```diff
@@ -0,0 +1,364 @@
+
+.. code-block:: js
+
+	driver.Connection.getTransaction(txCreateAliceSimpleSigned.id)
```
`conn.getTransaction`

TimDaub commented on docs/source/usage.rst:
```diff
@@ -0,0 +1,364 @@
+
+.. code-block:: js
+
+	conn.postTransaction(txTransferBobSigned, API_PATH)
```
Remove `API_PATH`

TimDaub commented on docs/source/usage.rst:
```diff
@@ -0,0 +1,364 @@
+
+.. code-block:: js
+
+	console.log('Is Bob the owner?', tx['outputs'][0]['public_keys'][0] == bob.publicKey)
```
`tx`is not defined anywhere.

michielmulders replied:
tx is defined in the code at bottom of document, this is just a snippet?
Would you do it differently?

TimDaub commented on docs/source/usage.rst:
```diff
@@ -0,0 +1,364 @@
+
+.. code-block:: js
+
+	let assets = conn.searchAssets('Bicycle Inc.')
```
`let assets` might never really get defined, as `searchAssets` returns a promise.

michielmulders replied:
Idd, removed.

TimDaub commented on docs/source/usage.rst:
```diff
@@ -0,0 +1,364 @@
+				0)
+			
+			// Sign with alice's private key
+			txTransferBobSigned = driver.Transaction.signTransaction(txTransferBob, alice.privateKey)
```
Needs `let`?

TimDaub replied:
This still isn't defined anywhere higher up, so needs a variable declaration? `let, const, var`?

michielmulders commented:
@kremalicious I changed the file permissions, can you confirm? @TimDaub can you do a review on the new commits. Changed the smaller stuff and added info about input/outputs (copied a bit from docs and source code).

TimDaub commented:
Merged via #99.

Great work @michielmulders 🎉
